### DISCUSSION TITLE
Can you stream Spotify to MPD?

### DISCUSSION THREAD

**Discussion by owl770** [MusicPlayerDaemon/MPD#1128] [Q&A] [answered]
Hi - Is there a way to connect MPD to Spotify (as a music source)? If so, are there any tutorials on it? I know Mopidy does it but I would like to keep using MPD. Thanks.

**Comment by MaxKellermann** [comment, marked as answer]
No, there's no way to do that. There used to be a plugin based on "libdespotify", but it was removed 6 years ago after the library became defunct.
Mopidy uses the officual but non-free "libspotify" library, which would be illegal to use with MPD (being GPL-licensed).
I personally don't care for Spotify support, because I don't have an account. The protocol is undocumented. If there ever is an effort to reverse-engineer the new protocol, and somebody interested submits some code, MPD may one day regain Spotify support, but don't hold your breath.

**Comment by owl770** [comment]
Thanks @MaxKellermann for the reply. I guess that explains why it doesn't show up in a google search!!!
I did find a VERY clunky way to get it there - librespot (from the raspotify package) piped to ffmpeg piped to cvlc to create an http stream that you can play via an MPD playlist (eg. http://servername:8080/spotify.mp3) but there was too much lag. When you paused a track on your phone, MPD continued to play for a few minutes (not an MPD issue but buffering in ffmpeg and cvlc).

If anyone is interested, the command is something like...

`/usr/bin/librespot --disable-audio-cache --name myspot --backend pipe | ffmpeg -re -f s16le -ar 44.1k -ac 2 -i pipe:0 -f mp3 - | cvlc - vvv input_stream --sout "#standard{access=http,mux=mp3,dst=:8080/spotify.mp3" --network-caching 50 --file-caching 50`

I ended up using librespot directly (bypassing MPD) to play to my USB amp directly via:

`/usr/bin/librespot --name myspot --device-type computer --backend alsa --bitrate 320 --disable-audio-cache --enable-volume-normalisation --volume-ctrl linear --initial-volume=25 --autoplay`

This creates a Spotify device (called myspot) that you can see on the Spotify app on your phone and outputs the music to the default alsa device on your RPi.

I guess this discussion can be closed.
